ca_tools.erl

```diff
@@ -0,0 +1,39 @@
+-module(ca_tools).
+-export([iterate/3, make_rule/1, make_state/1, random_state/1]).
+
+-spec iterate(ca:state(), [ca:rule()], non_neg_integer()) -> ca:state().
+iterate(State, _, 0) ->
+    State;
+iterate(State, Rules, N) ->
+    Generation = ca:process_state(State, Rules),
+    render(Generation),
+    iterate(Generation, Rules, N - 1).
+
+-spec render(ca:state()) -> string().
+render(S) ->
+    io:format("~p~n", [[render_cell(D) || D <- S]]).
+render_cell(Cell) when Cell == 0 -> $\s;
+render_cell(Cell) when Cell == 1 -> $#.
+
+-define(RULES, [[1,1,1],[1,1,0],[1,0,1],[1,0,0],[0,1,1],[0,1,0],[0,0,1],[0,0,0]]).
```

Comment:
Line 18 is too long: -define(RULES, [[1,1,1],[1,1,0],[1,0,1],[1,0,0],[0,1,1],[0,1,0],[0,0,1],[0,0,0]])..